plugins/destination/s3/client/write.go

```diff
@@ -133,6 +133,10 @@ func (c *Client) contentType() string {
 	if c.spec.ContentType != "" {
 		return c.spec.ContentType
 	}
+	if c.spec.Compression != "" {
```

Comment:
This assumes the only compression option in the filetypes package is `gzip` (which is currently correct) so ideally this should be a switch case. If `gzip`, do gzip, if not empty (but something else) leave it empty (unhandled) otherwise, decide based on Format.

Reply:
We could maybe put this in the filetypes spec, release that, and use it here.